xtask/src/dependencies.rs

```diff
@@ -0,0 +1,102 @@
+use std::{collections::HashMap, time::Instant};
+
+use crate::{
+    endgroup, group,
+    logging::init_logger,
+    utils::{
+        ensure_cargo_command_is_installed, format_duration, is_current_toolchain_nightly,
+        run_cargo, Params,
+    },
+};
+
+#[derive(clap::ValueEnum, Default, Copy, Clone, PartialEq, Eq)]
+pub(crate) enum DependencyCheckType {
```

Comment:
Not sure if we actually need `Type` at the end, just `DependencyCheck` might be a better name. I often use `Type`, or rather `Kind`, when there is an enum and a struct of the same name that interact.

Reply:
I removed the `Type` suffix.